client/web/src/global/CoolCodeIntel.tsx

```diff
@@ -180,18 +182,64 @@ export const ReferencesList: React.FunctionComponent<
         token: Token
     }
 > = props => {
-    const [activeLocation, setActiveLocation] = useState<Location>()
     const [filter, setFilter] = useState<string>()
     const debouncedFilter = useDebounce(filter, 150)
 
+    const {
+        lsifData,
+        error,
+        loading,
+        referencesHasNextPage,
+        implementationsHasNextPage,
+        fetchMoreReferences,
+        fetchMoreImplementations,
+        fetchMoreReferencesLoading,
+        fetchMoreImplementationsLoading,
+    } = usePreciseCodeIntel({
+        variables: {
+            repository: props.token.repoName,
+            commit: props.token.commitID,
+            path: props.token.filePath,
+            // On the backend the line/character are 0-indexed, but what we
+            // get from hoverifier is 1-indexed.
+            line: props.token.line - 1,
+            character: props.token.character - 1,
+            filter: filter || null,
+            firstReferences: 100,
+            afterReferences: null,
+            firstImplementations: 100,
+            afterImplementations: null,
+        },
+    })
+
+    const references = useMemo(() => (lsifData?.references.nodes ?? []).map(buildLocation), [lsifData])
+    const definitions = useMemo(() => (lsifData?.definitions.nodes ?? []).map(buildLocation), [lsifData])
+    const implementations = useMemo(() => (lsifData?.implementations.nodes ?? []).map(buildLocation), [lsifData])
+
+    // activeLocation is the location that is selected/clicked in the list of
+    // definitions/references/implementations.
+    const [activeLocation, setActiveLocation] = useState<Location>()
+    // We create an in-memory history here so we don't modify the browser
+    // location. This panel is detached from the URL state.
+    const blobMemoryHistory = useMemo(() => H.createMemoryHistory(), [])
+
+    // When the token for which we display data changed, we want to reset
+    // activeLocation & filter.
     useEffect(() => {
         setActiveLocation(undefined)
         setFilter(undefined)
     }, [props.token])
 
-    // We create an in-memory history here so we don't modify the browser
-    // location. This panel is detached from the URL state.
-    const blobMemoryHistory = useMemo(() => H.createMemoryHistory(), [])
+    // If props.jumpToFirst is true and we finished loading (and have
+    // definitions) we select the first definition. We set it as activeLocation
+    // and push it to the blobMemoryHistory so the code blob is open.
+    useEffect(() => {
+        if (props.jumpToFirst === true && definitions.length > 0) {
```

Comment:
```suggestion
        if (props.jumpToFirst && definitions.length > 0) {
```

As it's just `boolean | undefined` can we simplify